tests/test_style.py

```diff
@@ -485,9 +491,116 @@ def test_fetch_private_github_urls(tmp_path):
+        # Raw URL directly
+        "https://token@raw.githubusercontent.com/andreoliwa/nitpick/develop/initial.toml",
+    ],
+)
+def test_github_url_with_fixed_userinfo_token_has_correct_credential(url):
+    """Check private GitHub URLs with a token in various places are parsed correctly."""
+    parsed = GitHubURL.parse_url(url)
+    assert parsed.credentials == ("token", "")
+
+
+@pytest.mark.parametrize(
+    "url",
+    [
+        # Without commit reference (uses default branch)
+        "github://$TOKEN@andreoliwa/nitpick/initial.toml",
+        "gh://$TOKEN@andreoliwa/nitpick/initial.toml",
+        # Explicit commit reference
+        "github://$TOKEN@andreoliwa/nitpick@develop/initial.toml",
+        "gh://$TOKEN@andreoliwa/nitpick@develop/initial.toml",
+        # Regular GitHub URL
+        "https://$TOKEN@github.com/andreoliwa/nitpick/blob/develop/initial.toml",
+        # Raw URL directly
+        "https://$TOKEN@raw.githubusercontent.com/andreoliwa/nitpick/develop/initial.toml",
+    ],
+)
+def test_github_url_with_variable_userinfo_token_has_correct_credential(url, monkeypatch):
+    """Check private GitHub URLs with a token in various places are parsed correctly."""
+    monkeypatch.setenv("TOKEN", "envvar-token")
+    parsed = GitHubURL.parse_url(url)
+    assert parsed.credentials == ("envvar-token", "")
+
+
+@pytest.mark.parametrize(
+    "url",
+    [
+        # Without commit reference (uses default branch)
+        "github://andreoliwa/nitpick/initial.toml?token=$ENVVAR",
+        "gh://andreoliwa/nitpick/initial.toml?token=$ENVVAR",
+        # Explicit commit reference
+        "github://andreoliwa/nitpick@develop/initial.toml?token=$ENVVAR",
+        "gh://andreoliwa/nitpick@develop/initial.toml?token=$ENVVAR",
+        # Regular GitHub URL
+        "https://github.com/andreoliwa/nitpick/blob/develop/initial.toml?token=$ENVVAR",
+        # Raw URL directly
+        "https://raw.githubusercontent.com/andreoliwa/nitpick/develop/initial.toml?token=$ENVVAR",
+        # token in both userinfo and queryargs uses userinfo one
+        "github://$ENVVAR@andreoliwa/nitpick/initial.toml?token=$NOTUSED",
+    ],
+)
+def test_github_url_with_variable_query_token_has_correct_credential(url, monkeypatch):
+    """Check private GitHub URLs with a token in various places are parsed correctly."""
+    monkeypatch.setenv("ENVVAR", "envvar-token")
+    parsed = GitHubURL.parse_url(url)
+    assert parsed.credentials == ("envvar-token", "")
+
+
+def test_github_url_with_missing_envvar_has_empty_credential(monkeypatch):
+    """Environment var that doesn't exist is replaced with empty string."""
+    monkeypatch.delenv("MISSINGVAR", raising=False)
+    parsed = GitHubURL.parse_url("https://github.com/foo/bar/blob/branch/filename.toml?token=$MISSINGVAR")
+    assert parsed.credentials == ()
+
+
+@pytest.mark.xfail(reason="GithubURL currently doesnt preserve query args")
```

Comment:
From 49de33b728580ff7863a093e462dabd743dc5984: 
> includes an xfail test because currently querystrings aren't preserved
> when parsing github urls. Not sure if there are any practical scenarios
> where it matters, and test can just be removed as necessary.

This extra test does no harm, we can keep it. 👍🏻